js/index.js

```diff
@@ -1,42 +1,121 @@
+  const price = product.querySelector('.price span');
+  const quantity = product.querySelector('.quantity input');
+  
+  //console.log('Price is ' + price.innerHTML );
+  //console.log('Quantity selected is ' + quantity.value );
+  const subtotalNum = price.innerHTML*quantity.value;
+  //console.log(subtotalNum);
+  
+  //console.log(product.querySelector('.subtotal span').innerHTML)
+  product.querySelector('.subtotal span').innerHTML = subtotalNum; 
+  return subtotalNum;
+} 
 
 function calculateAll() {
+
+/*   
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
   const singleProduct = document.querySelector('.product');
   updateSubtotal(singleProduct);
-  // end of test
+  // end of test 
+*/
 
   // ITERATION 2
-  //... your code goes here
-
+  const allProductsHtmlColl = document.getElementsByClassName('product')
+  //console.log(allProductsHtmlColl);
+  const allProducts = [...allProductsHtmlColl];
+  //console.log(allProducts);
+  const allSubtotals = [];
+  allProducts.forEach( product => {
+    allSubtotals.push(updateSubtotal(product))
+  })
+  
+  
   // ITERATION 3
-  //... your code goes here
+  const total = allSubtotals.reduce(
+    (accumulator, currentValue) => accumulator + currentValue,
+    0
+  );
+  document.querySelector('#total-value span').innerHTML = total;
+
 }
 
+
 // ITERATION 4
 
 function removeProduct(event) {
   const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+  //console.log('The target in remove is:', target);
+  const rowToRemove = target.parentNode.parentNode;
+  const parentOfRowToRemove = rowToRemove.parentNode;
+  //console.log('The element to remove is:', rowToRemove);
+  //console.log('The parent element is:', parentOfRowToRemove);
+  parentOfRowToRemove.removeChild(rowToRemove);
 }
 
+
 // ITERATION 5
 
-function createProduct() {
-  //... your code goes here
+function createProduct(event) {
+  const target = event.currentTarget;
+  //console.log('The target in create is:', target);
+  const infoOfRowToAdd = target.parentNode.parentNode.parentNode;
+  const parentOfRowToAdd = document.querySelector('tbody');
+  //console.log('The parent of the info is:', infoOfRowToAdd);
+  //console.log('The parent of the row to add is:', parentOfRowToAdd);
+
+  const name = infoOfRowToAdd.querySelector('.name input').value;
+  const price = infoOfRowToAdd.querySelector('.price input').value;
+  //console.log(name);
+  //console.log(price);
+
+
+    // ---> Approach with clone node 
+
+  let oneRow = document.querySelector('.product');
+  //console.log(oneRow);
+  let clonedNode = oneRow.cloneNode(true);
+  parentOfRowToAdd.appendChild(clonedNode);
+  clonedNode.setAttribute("id", "newProduct1");
+  document.querySelector('#newProduct1 :nth-child(1)').innerHTML=name;
+  document.querySelector('#newProduct1 :nth-child(2)').innerHTML=price;
+
```

Comment:
You're very close here to the solution! 

- Remember, that in html you have input fields to enter product name and price.  -> this is the value you want to use
- After you clone the node , you want to set its innerHtml to that value you read from the input
```
clonedNode.querySelector('.name span').innerHTML = productName.value; 
```
`productName` is variable you have to create beforehand, for example:
```
const productName = document.querySelectorAll('.create-product input')
```